packages-exp/auth-exp/src/core/user/reload.ts

```diff
@@ -46,6 +46,13 @@ export async function _reloadWithoutSaving(user: User): Promise<void> {
   const newProviderData = coreAccount.providerUserInfo?.length
     ? extractProviderData(coreAccount.providerUserInfo)
     : [];
+
+  const providerData = mergeProviderData(user.providerData, newProviderData);
+
+  const oldIsAnonymous = user.isAnonymous;
```

Comment:
can you add some comments please? why do we need all this logic to figure out if the user is anonymous?

Reply:
Done